medusa/storage/s3_storage.py

```diff
@@ -56,22 +57,24 @@ def connect_storage(self):
             driver = cls(profile['aws_access_key_id'], profile['aws_secret_access_key'])
             return driver
 
-    def download_blobs(self, src, dest):
+    def download_blobs(self, srcs, dest):
         """
         Downloads a list of files from the remote storage system to the local storage
 
         :param src: a list of files to download from the remote storage system
         :param dest: the path where to download the objects locally
         :return:
         """
-        for src_obj in list(src):
+        for src_obj in list(srcs):
             blob = self.get_blob(src_obj)
             index = src_obj.rfind('/')
             if index > 0:
                 file_name = src_obj[src_obj.rfind('/') + 1:]
             else:
                 file_name = src_obj
-            blob.download(os.path.join(dest, file_name), overwrite_existing=True)
+            src_path = Path(src_obj)
+            blob_dest = '{}/{}'.format(dest, src_path.parent.name) if src_path.parent.name.startswith('.') else dest
```

Comment:
This conditional assignment is used in 3 or 4 different places now, maybe we could make it a function to avoid duplication?
It's not a blocker though.

Reply:
They do look the same, but there are tiny differences here and there:

In gsutil/_dowload paths:
`        new_dest = '{}/{}'.format(old_dest, parent) if parent.startswith('.') else old_dest`
In gsutil/_upload paths:
`        new_dest = '{}/{}'.format(old_dest, parent) if parent.startswith('.') else old_dest`

In concurrent/__upload_file:
`    obj_name = '{}/{}'.format(src.parent.name, src.name) if src.parent.name.startswith('.') else src.name`
In concurrent/__download_file:
`        blob_dest = '{}/{}'.format(dest, src_path.parent.name) if src_path.parent.name.startswith('.') else dest`

In s3_storage/download_blobs:
`            blob_dest = '{}/{}'.format(dest, src_path.parent.name) if src_path.parent.name.startswith('.') else dest`

I also like extracting shared code, but this case might not be worth it.